File: autumn/human-readable-number/js/log10.js
function numberFormat(d) {
   let e = Math.trunc(Math.log10(d) / 3);
   return (d / 1e3 ** e).toFixed(3) + ['', ' k', ' M', ' G'][e];
}

let s = numberFormat(9012345678);
console.log(s == '9.012 G');

File: autumn/human-readable-number/nim/log10.nim
from math import nil
from strformat import fmt

proc numberFormat(d: float): string =
   let e = math.trunc(math.log10(d)) / 3
   let f = d / math.pow(1e3, e)
   return fmt"{f:.3f}" & ["", " k", " M", " G"][int e]

let s = numberFormat(9012345678.0)
echo s == "9.012 G"
